loopy/transform/callable.py

```diff
@@ -154,6 +154,81 @@ def __call__(self, target, identifier):
         return None
 
 
+def subarrayrefs_are_equiv(sar1, sar2, knl):
+    """
+    Compares if two instance of :class:`loopy.symbolic.SubArrayRef`s point
+    to the same array region.
+    """
+    from loopy.kernel.function_interface import get_arg_descriptor_for_expression
+
+    return get_arg_descriptor_for_expression(knl, sar1) == (
+            get_arg_descriptor_for_expression(knl, sar2)) and (
+                    sar1.get_begin_subscript(knl) ==
+                    sar2.get_begin_subscript(knl))
+
+
+def _check_correctness_of_args_and_assignees(insn, callee_kernel, caller_knl):
```

Comment:
- [ ] Add a docstring to describe what's actually being checked.
- [ ] Add section comments `{{{` to describe each sub-check bit.

Reply:
Done in 71b05d5b.